docs/index.rst

```diff
@@ -0,0 +1,7 @@
+Envoy v2 API documentation
```

Comment:
Can we auto generate the index also? Or at least put TODO for follow up hacking?